File: smart_lock_library/src/main/java/smart_lock_library/preference/SmartLockLibraryPreference.java
package smart_lock_library.preference;

import android.content.Context;

import core.preference.SharedPreferencesHelper;
import smart_lock_library.utils.SmartLockConstants;

public class SmartLockLibraryPreference extends SharedPreferencesHelper {

    public static void setUserRefusedSmartLock(Context context) {
        SharedPreferencesHelper.setBooleanPreference(context, SmartLockConstants.PREFERENCE_KEY_USE_SMART_LOCK, false);
    }

    public static boolean shouldUseSmartLock(Context context) {
        return SharedPreferencesHelper.getBooleanPreference(context, SmartLockConstants.PREFERENCE_KEY_USE_SMART_LOCK);
    }

    public static void resetUseSmartLock(Context context) {
        SharedPreferencesHelper.resetSinglePreference(context, SmartLockConstants.PREFERENCE_KEY_USE_SMART_LOCK);
    }
}

File: app/src/main/java/smart_lock_library_kt/preference/SmartLockLibraryPreference.kt
package smart_lock_library_kt.preference

import android.content.Context
import core.preference.SharedPreferencesHelper
import smart_lock_library_kt.utils.PREFERENCE_KEY_USE_SMART_LOCK

class SmartLockLibraryPreference : SharedPreferencesHelper() {

    fun setUserRefusedSmartLock(context: Context) {
        SharedPreferencesHelper.setBooleanPreference(context, PREFERENCE_KEY_USE_SMART_LOCK, false)
    }

    fun shouldUseSmartLock(context: Context): Boolean {
        return SharedPreferencesHelper.getBooleanPreference(context, PREFERENCE_KEY_USE_SMART_LOCK)
    }

    fun resetUseSmartLock(context: Context) {
        SharedPreferencesHelper.resetSinglePreference(context, PREFERENCE_KEY_USE_SMART_LOCK)
    }
}